rm-datasource/src/main/java/io/seata/rm/datasource/exec/SelectForUpdateExecutor.java

```diff
@@ -60,9 +67,17 @@ public T doExecute(Object... args) throws Throwable {
         String selectPKSQL = buildSelectSQL(paramAppenderList);
         try {
             if (originalAutoCommit) {
+                /*
+                 * prevent release local lock before get global lock
+                 * dose not need the savepoint if original auto commit was true
```

Comment:
In order to hold the local db lock during global lock checking, set auto commit value to false first if original auto commit was true?

Reply:
ok